indra/newview/lldrawpoolmaterials.cpp

```diff
@@ -245,23 +247,11 @@ void LLDrawPoolMaterials::renderDeferred(S32 pass)
         }
 
         // upload matrix palette to shader
-        if (rigged && params.mAvatar.notNull())
+        if (rigged)
         {
-            if (params.mAvatar != lastAvatar)
+            if (!uploadMatrixPalette(params.mAvatar, params.mSkinInfo, lastAvatar, lastMeshId, skipLastSkin))
             {
-                const LLVOAvatar::MatrixPaletteCache& mpc = params.mAvatar->updateSkinInfoMatrixPalette(params.mSkinInfo);
-                U32 count = static_cast<U32>(mpc.mMatrixPalette.size());
-
-                if (count == 0)
-                {
-                    //skin info not loaded yet, don't render
-                    return;
-                }
-
-                mShader->uniformMatrix3x4fv(LLViewerShaderMgr::AVATAR_MATRIX,
-                    count,
-                    false,
-                    (GLfloat*)&(mpc.mGLMp[0]));
+                continue;
```

Comment:
This is now a `continue`, but was previously a `return`. The `return` appeared to not be correct, as it would interfere with the rendering of other objects in the draw pool.